llms/llama/llama.py

```diff
@@ -326,28 +331,36 @@ def generate(question):
         print()
 
 
+def sanitize_config(config, weights):
+    config.pop("model_type", None)
+    n_heads = config["n_heads"]
+    if "n_kv_heads" not in config:
+        config["n_kv_heads"] = n_heads
+    if "head_dim" not in config:
+        config["head_dim"] = config["dim"] // n_heads
+    if "hidden_dim" not in config:
+        config["hidden_dim"] = weights["layers.0.feed_forward.w1.weight"].shape[0]
+    if config.get("vocab_size", -1) < 0:
+        config["vocab_size"] = weights["output.weight"].shape[-1]
+    if "rope_theta" not in config:
+        config["rope_theta"] = 10000
+    unused = ["multiple_of", "ffn_dim_multiplier"]
+    for k in unused:
+        config.pop(k, None)
+    return config
```

Comment:
I don't like the need for this function. I think the way to fix it is to do this mapping once in the conversion script and save the updated `config.json`. Unfortunately that would require updating all the Hugging Face models which is a bit tedious :\

Reply:
FWIW that's the approach taken by the other examples, so they avoid this baggage.